src/ImageSharp/Common/Helpers/SimdUtils.HwIntrinsics.cs

```diff
@@ -230,22 +328,183 @@ public static class HwIntrinsics
                         ref Vector128<byte> vs0 = ref Unsafe.Add(ref sourceBase, i);
                         ref Vector128<byte> vd0 = ref Unsafe.Add(ref destBase, i);
 
-                        vd0 = Ssse3.Shuffle(vs0, vcm);
-                        Unsafe.Add(ref vd0, 1) = Ssse3.Shuffle(Unsafe.Add(ref vs0, 1), vcm);
-                        Unsafe.Add(ref vd0, 2) = Ssse3.Shuffle(Unsafe.Add(ref vs0, 2), vcm);
-                        Unsafe.Add(ref vd0, 3) = Ssse3.Shuffle(Unsafe.Add(ref vs0, 3), vcm);
+                        vd0 = Ssse3.Shuffle(vs0, vshuffle);
+                        Unsafe.Add(ref vd0, 1) = Ssse3.Shuffle(Unsafe.Add(ref vs0, 1), vshuffle);
+                        Unsafe.Add(ref vd0, 2) = Ssse3.Shuffle(Unsafe.Add(ref vs0, 2), vshuffle);
+                        Unsafe.Add(ref vd0, 3) = Ssse3.Shuffle(Unsafe.Add(ref vs0, 3), vshuffle);
                     }
 
                     if (m > 0)
                     {
                         for (int i = u; i < n; i++)
                         {
-                            Unsafe.Add(ref destBase, i) = Ssse3.Shuffle(Unsafe.Add(ref sourceBase, i), vcm);
+                            Unsafe.Add(ref destBase, i) = Ssse3.Shuffle(Unsafe.Add(ref sourceBase, i), vshuffle);
                         }
                     }
                 }
             }
 
+            [MethodImpl(InliningOptions.ShortMethod)]
+            private static void Shuffle3(
+                ReadOnlySpan<byte> source,
+                Span<byte> dest,
+                byte control)
+            {
+                if (Ssse3.IsSupported)
```

Comment:
I guess there is no theoretical blocker for processing 2x more elements with a single step by AVX2?

Reply:
Looks like this one is expanding to 3->4 channels so it can shuffle 4 and then going back to 3?  It likely wouldn't be faster with all the extra permutes required to implement it with AVX2.  The code that fills the 4th channel with 0xFF could be removed, though... that channel gets thrown away after the shuffle anyway.

Reply:
Oh yeah, I forgot about that. I don't need it now I've limited the indices to `0,1,2` for this method.